### DISCUSSION TITLE
Popover and following its anchor on scroll

### DISCUSSION THREAD

**Discussion by Zerebokep** [radix-ui/primitives#2681] [Help] [answered]
Is there any way to stop the popover from following its anchor when I scroll? I would like to let it stay at the position it opened.

**Comment by Zerebokep** [comment]
push

**Comment by benoitgrelard** [comment, marked as answer]
Hi @Zerebokep, this isn't currently supported.
I would like to learn more about your use case to see if it would be worth adding an option to achieve it.
Can you explain a bit more what you are trying to achieve?

**Reply by Zerebokep** [reply]
Sure, I have a chat where a user can trigger a popover by clicking on a username, now when the popover is active and the user scrolls, the popover vanishes. I would like to replicate the behavior like in Discord (popover keeps at its start position).

**Reply by benoitgrelard** [reply]
Can you show me what that looks like in Discord as you mention this as an example?

**Reply by Zerebokep** [reply]
Yes, check this video (expires in 2 days): https://streamable.com/qizfy6

**Reply by benoitgrelard** [reply]
Interesting, this is a `HoverCard` rather than a `Popover` but I guess the idea is the same.

I could see it somehow make sense with a `HoverCard` but for some reason when I see it for a `Popover` it just feels like a bug to me as a user as it feels super unexpected.

Discord actually does it and it looks broken to me because you've lost context at this point, you don't know on which post you're acting:

https://github.com/radix-ui/primitives/assets/1539897/054c9a23-1f6f-4708-bc14-f8f3a65fe7e7

**Reply by Zerebokep** [reply]
That's true, maybe it's just because I'm used to it from Discord, but it feels more broken if it follows the anchor. 😁 

Initially I thought the Popover sticky option was exactly for that.

edit: after testing both versions I agree with you, it's better to keep it connected to the anchor on scroll

**Reply by benoitgrelard** [reply]
It is, but only really works on the main axis and we use the limiter so it doesn't stay stuck in the view whilst the trigger is not in view anymore.

At the moment this isn't something we are able to accommodate unfortunately as we try to keep the API surface small and simple for most use-case. It's a difficult balance to achieve and we may need to allow more behaviours at some point.
